virttest/utils_test/__init__.py

```diff
@@ -1755,46 +1755,49 @@ def __str__(self):
-            _check_cmds = {'unixbench': "pidof -s ./Run",
-                           'stress': "pidof -s stress",
-                           'iozone': "pidof -s iozone"}
-            _stop_cmds = {'unixbench': "killall ./Run",
-                          'stress': "killall stress",
-                          'iozone': "killall iozone"}
-            try:
-                control_file = _control_files[stress_type]
-                self.control_path = os.path.join(data_dir.get_root_dir(),
-                                                 "shared/control",
-                                                 control_file)
-                self.check_cmd = _check_cmds[stress_type]
-                self.stop_cmd = _stop_cmds[stress_type]
-            except KeyError:
-                self.control_path = ""
-                self.check_cmd = ""
-                self.stop_cmd = ""
+        :param vm: the vm to be loading stress
+        :param stress_type: the name of the stress tool
+        :param download_url: from where download the stress tool
+        :param make_cmds: make command of the stress tool
+        :param stress_cmds: the command to launch stress,
+        use stress_type instead if not defined
+        :param stress_args: the arguments of the stress tool
+        :param work_path: the relative work path of the stress tool,
+        e.g. for iozone: "src/current"
+        :param check_installed_cmds: the command to check if the tool is installed,
+        skip the check if not defined.
+        :param download_type: currently support "git" or "tarball"
+        """
 
         self.vm = vm
         self.params = vm.params
         self.timeout = 60
         self.stress_type = stress_type
-        if stress_type not in ["stress", "unixbench", "iozone"]:
-            raise StressError("Stress %s is not supported now." % stress_type)
-
-        _parameters_filter(stress_type)
-        self.stress_args = self.params.get("stress_args", "")
+        stress_cmds = stress_cmds or stress_type
+        self.stress_cmds = self.params.get('stress_cmds_%s' % stress_type, stress_cmds)
+        self.stress_args = self.params.get("%s_args" % stress_type, stress_args)
+        self.download_url = self.params.get('download_url_%s' % stress_type,
+                                            download_url)
+        self.download_type = download_type
+        self.base_name = self.download_url.split("/")[-1]
+        self.make_cmds = self.params.get('make_cmds_%s' % stress_type, make_cmds)
+        self.work_path = self.params.get('%s_work_path' % stress_type, work_path)
+        self.check_cmd = "pidof -s %s" % stress_type
+        self.stop_cmd = "killall %s" % stress_type
+        self.dst_path = self.params.get('stress_dst_path', '/home')
+        self.check_installed_cmds = self.params.get('check_installed_cmds_%s' %
+                                                    stress_type, check_installed_cmds)
+        self.session = self.get_session()
```

Comment:
why to use self.session here, self.get_session() returns Bool ?

Reply:
@balamuruhans Every time I call self.get_session(), it will open up a new session and return it. I previously use separated sessions in each functions by calling get_session() and close it when the function finished its task. But this will lead to massive session logs in test result folder, which is hard to debug. So I only open up one session in the beginning, and use this one session for all functions, and close it until clean() is called. Actually I am not really sure which method is better...

Reply:
@balamuruhans It return session in line 1805. But I agree that closing the session immediately after using it will be safer, I will modify it.